plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/plan/CubridPlanNode.java

```diff
@@ -0,0 +1,177 @@
+
+    public CubridPlanNode(@NotNull String queryPlan) {
+        this(null, null, null, queryPlan);
+    }
+
+    private CubridPlanNode(@Nullable CubridPlanNode parent, @Nullable String name, @Nullable List<String> segments, @NotNull String fullText) {
+        this.parent = parent;
+        this.name = name;
+        this.fullText = fullText;
+        parseObject(parent == null ? this.getSegments() : segments);
+        parseNode();
+    }
+
+    @NotNull
+    @Property(order = 0, viewable = true)
+    @Override
+    public String getNodeType() {
+        return getMethodTitle(name);
+    }
+
+    @NotNull
+    @Property(order = 1, viewable = true)
+    @Override
+    public String getNodeName() {
+        return classNode.get(nodeName);
+    }
+    
+    @NotNull
+    @Property(order = 2, viewable = true)
+    public Number getCost() {
+        return cost;
+    }
+    
+    @NotNull
+    @Property(order = 3, viewable = true)
+    public Number getCardinality() {
+        return row;
+    }
+    
+    @NotNull
+    @Property(order = 4, length = PropertyLength.MULTILINE)
+    public String getFullText() {
+        return fullText;
+    }
+    
+    @Nullable
+    @Override
+    public CubridPlanNode getParent() {
+        return parent;
+    }
+    
+    @Nullable
+    @Override
+    public Collection<CubridPlanNode> getNested() {
+        return nested;
+    }
+
+    private String getMethodTitle(@NotNull String method) {
+
+        return switch (method) {
+            case "iscan" -> "Index Scan";
+            case "sscan" -> "Full Scan";
+            case "temp(group by)" -> "Group by Temp";
+            case "temp(order by)" -> "Order by Temp";
+            case "nl-join (inner join)" -> "Nested Loop - Inner Join";
+            case "nl-join (cross join)" -> "Nested Loop - Cross Join";
+            case "idx-join (inner join)" -> "Index Join - Inner Join";
+            case "m-join (inner join)" -> "Merged - Inner Join";
+            case "temp" -> "Temp";
+            case "follow" -> "Follow";
+            default -> method;
+        };
+    }
+
+    private void addNested(@NotNull String name, @NotNull List<String> value) {
+        if (nested == null) {
+            nested = new ArrayList<>();
+        }
+        nested.add(new CubridPlanNode(this, name, value, fullText));
+    }
+
+    private void parseNode() {
+        for (String key : nodeProps.keySet()) {
+            if (key.contains("class")) {
+                this.nodeName = nodeProps.get("class").split(" ")[1];
+            } else if (key.equals("cost")) {
+                String[] values = nodeProps.get(key).split(" card ");
+                this.cost = Integer.parseInt(values[0]);
+                this.row = Integer.parseInt(values[1]);
+            }
+        }
+    }
+
+    private void parseObject(@NotNull List<String> segments) {
+        if (!segments.isEmpty()) {
+            String[] removes = segments.remove(0).split(":");
```

Comment:
Could you please move all hardcoded strings with duplicates into constants?
Like the:
`private static final String OPTIONS_SEPARATOR = ":";`